### PR TITLE
Adding JS/TS templates for cart/checkout validations

### PR DESCRIPTION
**Pull request by cpeddecord**
Adds JS/TS support for cart/checkout validation

(Generally) follow instructions from the [workshop](https://workshops.shopify.dev/workshops/javascript-functions-preview#0), but generate a cart/checkout validation instead.

```
npm init @shopify/app@pre --package-manager=yarn
```
```
yarn generate extension --type cart_checkout_validation --clone-url https://github.com/Shopify/function-examples#cart-validation-function-template
```

### CODE REVIEW

File: .graphqlrc.cjs [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
 
-    const extensions = fs.readdirSync(`./${path}`);
-    for (const entry of extensions) {
-        const extensionPath = `./${path}/${entry}`;
-        const schema = `${extensionPath}/schema.graphql`;
-        if(!fs.existsSync(schema)) {
-            continue;
-        }
-
-        const projectName = extensionPath.substring(2).replaceAll('/', '-');
```
Comment (aledalgrande) [inline .graphqlrc.cjs:17]: might want to drop these format changes, to make the diff cleaner

File: checkout/javascript/cart-checkout-validation/src/index.liquid [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{%- if flavor contains "vanilla-js" -%}
```
Comment (aledalgrande) [inline checkout/javascript/cart-checkout-validation/src/index.liquid:1]: neat

**Review by aledalgrande — APPROVED** [review]
Mind adding a `.gitignore` as well?

File: checkout/javascript/cart-checkout-validation/src/index.liquid [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+/**
+ * @type {FunctionResult}
+ */
+export default /**
+ * @param {InputQuery} input
+ * @returns {FunctionResult}
+ */
+function cartCheckoutValidation(input) {
```
Comment (nickwesselman) [inline checkout/javascript/cart-checkout-validation/src/index.liquid:16]: Maybe be consistent with other templates here and use an anonymous arrow function instead?

File: checkout/javascript/cart-checkout-validation/src/index.liquid [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const errors = []
```
Comment (nickwesselman) [inline checkout/javascript/cart-checkout-validation/src/index.liquid:17]: Should be able to simplify this with something like `return car.lines.filter(<quantity greater than 1>).map(<error>)`

File: checkout/javascript/cart-checkout-validation/src/index.liquid [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+};
+{%- elsif flavor contains "typescript" -%}
+import {
+  InputQuery,
+  FunctionResult,
+  FunctionError,
+} from "../generated/api";
+
+export default function cartCheckoutValidation(input: InputQuery): FunctionResult {
+  const errors: FunctionError = []
```
Comment (nickwesselman) [inline checkout/javascript/cart-checkout-validation/src/index.liquid:40]: Same filter/map comment here

**Review by nickwesselman — APPROVED** [review]
Awesome, thanks for creating this! Couple nits on the JS code. I can tophat tomorrow as well if you need it.

**Pull request merged**